### PR TITLE
Use multiprecision for `Constant` 

### PR DESCRIPTION
**Pull request by ajay-mk**
Address #101

### CODE REVIEW

File: SeQuant/core/rational.hpp [hunk 1]
```diff
@@ -33,12 +34,31 @@ std::wstring to_wstring(const boost::rational<T>& i) {
...
+
+// introducing multiprecision
+using namespace boost::multiprecision;
+
+// typedef boost::multiprecision::number<
+//     cpp_int_backend<0, 0, signed_magnitude, checked, void>>
+//     mp_int_type;
+
+// try high fixed precision
+typedef int512_t mp_int_type;
```
Comment (Krzmbrzl) [inline SeQuant/core/rational.hpp:46]:
```suggestion
using int512_t = mp_int_type;
```
(that's the more c++/modern way of doing it)

File: SeQuant/core/rational.hpp [hunk 2]
```diff
@@ -33,12 +34,31 @@ std::wstring to_wstring(const boost::rational<T>& i) {
...
+
+using rational = boost::rational<mp_int_type>;
+
+// using rational = boost::rational<std::int64_t>;
+
```
Comment (Krzmbrzl) [inline SeQuant/core/rational.hpp:51]:
```suggestion
```

File: SeQuant/core/rational.hpp [hunk 3]
```diff
@@ -33,12 +34,31 @@ std::wstring to_wstring(const boost::rational<T>& i) {
...
 
 namespace sequant {
-using rational = boost::rational<std::int64_t>;
+
+// introducing multiprecision
+using namespace boost::multiprecision;
+
+// typedef boost::multiprecision::number<
+//     cpp_int_backend<0, 0, signed_magnitude, checked, void>>
+//     mp_int_type;
```
Comment (Krzmbrzl) [inline SeQuant/core/rational.hpp:43]:
```suggestion
```

File: SeQuant/core/rational.hpp [hunk 4]
```diff
@@ -33,12 +34,31 @@ std::wstring to_wstring(const boost::rational<T>& i) {
...
+
+// try high fixed precision
```
Comment (Krzmbrzl) [inline SeQuant/core/rational.hpp:45]: What's this comment supposed to mean?

**Comment by ajay-mk** [comment]
Thanks for the comments @Krzmbrzl.

**Comment by Krzmbrzl** [comment]
You're welcome :)

**Comment by evaleev** [comment]
@ajay-mk I think this is good to go now ... fix the factorial issue in another PR

**Review by evaleev — APPROVED** [review]

**Pull request merged**